### ISSUE TITLE
Missing Csocket source in 1.6.1 published tarball

### ISSUE THREAD

**Issue by triplepoint** [znc/znc#1175]
I might be misunderstanding the ZNC build process, but it seems that the `third_party\Csocket` directory is a git submodule, and that content is not making it into the published tarball at http://znc.in/releases/znc-1.6.1.tar.gz.

This manifests at build time, I think, as something like:
```
...
Building core object Csocket...
src/Csocket.cpp: In member function ‘virtual bool Csock::SSLClientSetup()’:
src/Csocket.cpp:1467:48: error: ‘SSLv3_client_method’ was not declared in this scope
   m_ssl_ctx = SSL_CTX_new( SSLv3_client_method() );
                                                ^
src/Csocket.cpp: In member function ‘SSL_CTX* Csock::SetupServerCTX()’:
src/Csocket.cpp:1589:43: error: ‘SSLv3_server_method’ was not declared in this scope
   pCTX = SSL_CTX_new( SSLv3_server_method() );
```

Maybe I misunderstood what's happening at build time, and there's something subtle about that `third_party` directory in the distributed tarball?

In any case, I've built an Ubuntu 14.04 virtual machine, and followed the instructions at http://wiki.znc.in/Installation#Build_From_Source and received the above error at make time.

I believe this same process worked fine about a month ago, if that's of any help.

**Comment by DarthGandalf** [comment]
When packing tarball, `Csocket.cc` from the submodule is copied to `src/Csocket.cpp` with some modifications (see `make-tarball.sh`). Yes, it's subtle.

The SSL error is fixed at #1146 so you can use 1.7 from either git or nightly tarball

**Comment by triplepoint** [comment]
A-ha, so I was barking up the wrong tree.  Thanks for the quick feedback, I'll try the nightlies.
